CMSgov/mint-app
# [EASI-3886] - Unit test warning fixes

# EASI-3886

## Changes and Description

- Addressed all warning related to missing gql fields in mocked data
- Updated snapshots

## Outstanding Warnings (Reasons noted)

- Warning: validateDOMNesting(...): `<li>` cannot appear as a descendant of `<li>`
  - This one is curious as I've verified there is no direct DOM nesting of `<li>`.  According to documentation, nesting is permissible if the `<li>` elements are wrapped in a `<ul>`, like so
  ```<ul>
        <li>DSA</li>
        <ul>
            <li>Array</li>
            <li>Linked List</li>
            <li>stack</li>
            <li>Queue</li>
        </ul>
        <li>Web Technologies</li>
        <ul>
            <li>HTML</li>
            <li>CSS</li>
            <li>JavaScript</li>
        </ul>
        <li>Aptitude</li>
        <li>Gate</li>
        <li>Placement</li>
    </ul>```
[https://www.geeksforgeeks.org/what-is-nesting-of-list-how-to-create-the-nested-list-in-html/](url)
- Warning: Can't perform a React state update on an unmounted component
  -  I'm going to leave this as a warning, as it doesn't necessarily indicate an issue, and will be removed/address in React 18
  [https://stackoverflow.com/questions/72028010/cant-perform-a-react-state-update-on-an-unmounted-component-in-react-testing-li](url)
- Cache data may be lost when replacing the modelPlan field of a Query object
  - We have 2 gql entities/typeNames that have no unique identifier, and really shouldn't be cached.  I think it's fine to ignore these warnings for now, but might still look to see if we can explicitly disable these
  `DiscussionRoleSelection`
  `PrepareForClearance`



## How to test this change

Run the unit tests suite - `scripts/dev test:js` and verify the tests pass and no warnings outside of the ones noted above

## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
